lib/pages/profile.dart

```diff
@@ -8,11 +11,38 @@ class Profile extends StatelessWidget {
 
     return Scaffold(
       body: Center(
-        child:Text(getString(context, "profile"))
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Text(getString(context, "profile")),
+            RaisedButton(
+              key: Key("debug_delete_account_button"),
+              color: Colors.red,
+              onPressed: () => deleteAccount(UserService().user, context),
+              child: Text(getString(context, 'debug_delete_account_button'))
+            ),
+          ],
+        ),
       )
-
     );
+  }
 
+  void openAuthenticationPage(context) {
+    Navigator.of(context).pushNamed('/');
```

Comment:
you should cleanup the navigation stack when returnin to the auth page

try  Navigator.popUntil(context, ModalRoute.withName('/'));